### PR TITLE
PR: Improve check to see if Spyder is running in a macOS app

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes
Changes needed for #10054 to work

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

File: spyder/config/base.py [hunk 1]
```diff
@@ -500,8 +500,11 @@ def translate_gettext(x):
 else:
     MAC_APP_NAME = 'Spyder-Py2.app'
 
-def running_in_mac_app():
-    if sys.platform == "darwin" and MAC_APP_NAME in __file__:
+
+def running_in_mac_app(check_file=False):
```
Comment (ccordoba12) [inline spyder/config/base.py]: Please add a docstring to this function and describe what `check_file` is used for.

File: spyder/app/mainwindow.py [hunk 2]
```diff
@@ -1320,7 +1320,7 @@ def post_visible_setup(self):
         # In MacOS X 10.7 our app is not displayed after initialized (I don't
         # know why because this doesn't happen when started from the terminal),
         # so we need to resort to this hack to make it appear.
-        if running_in_mac_app():
+        if running_in_mac_app(True):
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: Please use `check_file=True` here.

File: spyder/app/start.py [hunk 3]
```diff
@@ -143,7 +143,7 @@ def main():
         os.environ['SPYDER_DEBUG'] = levels[options.debug_info]
 
     if CONF.get('main', 'single_instance') and not options.new_instance \
-      and not options.reset_config_files and not running_in_mac_app():
+      and not options.reset_config_files and not running_in_mac_app(True):
```
Comment (ccordoba12) [inline spyder/app/start.py]: Please use `check_file=True` here.

File: spyder/preferences/general.py [hunk 4]
```diff
@@ -69,7 +69,7 @@ def setup_page(self):
                               'check_updates_on_startup')
 
         # Decide if it's possible to activate or not single instance mode
-        if running_in_mac_app():
+        if running_in_mac_app(True):
```
Comment (ccordoba12) [inline spyder/preferences/general.py]: Please use `check_file=True` here.

File: spyder/config/base.py [hunk 5]
```diff
@@ -500,8 +500,19 @@ def translate_gettext(x):
 else:
     MAC_APP_NAME = 'Spyder-Py2.app'
 
-def running_in_mac_app():
-    if sys.platform == "darwin" and MAC_APP_NAME in __file__:
+
+def running_in_mac_app(check_file=False):
+    """
+    Check if the app is running in mac.
```
Comment (ccordoba12) [inline spyder/config/base.py]: Please change this to

    Check if Spyder is running inside an app on macOS.

**Comment by impact27** [comment]
Could you retrigger the tests?

**Pull request merged**